src/x86/uarch.c

```diff
@@ -2,6 +2,12 @@
 
 #include <cpuinfo.h>
 #include <x86/api.h>
+#include <x86/cpuid.h>
+
+CPUINFO_INTERNAL bool cpuinfo_x86_detect_pcores(){
```

Comment:
Will this work on pcore-only CPUs?  I'm thinking the reverse logic should be used?
cpuinfo_x86_detect_hybrid_ecore(){
If the hybrid is not present, pcore should be assumed.

Reply:
Thank you, we have modified it to check whether it is a hybrid architecture.
```
CPUINFO_INTERNAL bool cpuinfo_x86_detect_hybrid_ecore(){
	if (((cpuid(0x1A).eax >> 24) & 0xFFF) == 0x20) return true;
	return false;
}
```